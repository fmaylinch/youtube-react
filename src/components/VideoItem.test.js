//@flow

import React from 'react'
import { shallow } from 'enzyme'
import VideoItem from '../components/VideoItem'
import type { Video } from '../components/types.js'

describe('VideoItem', () => {

	const video: Video = {
		id: "4TPqUvy1vYU",
		image: {
			width: 320,
			height: 180,
			url: "https://i.ytimg.com/vi/4TPqUvy1vYU/mqdefault.jpg"
		},
		title: "St. Vincent - New York (Official Video)"
	}

	it('renders component correctly', () => {

    const component = shallow(<VideoItem video={video}/>)
		expect(component).toMatchSnapshot()
	})

	it('renders image correctly', () => {

    const component = shallow(<VideoItem video={video}/>)
		const image = <img src={video.image.url} alt={video.title} />
		expect(component.contains(image)).toBeTruthy()
	})
})
