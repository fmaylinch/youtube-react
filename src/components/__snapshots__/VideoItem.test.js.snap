// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem renders component correctly 1`] = `
<div>
  <img
    alt="St. Vincent - New York (Official Video)"
    src="https://i.ytimg.com/vi/4TPqUvy1vYU/mqdefault.jpg"
  />
</div>
`;
